trace/kafka/component.go

```diff
@@ -0,0 +1,158 @@
+package kafka
+
+import (
+	"errors"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/beatlabs/patron/encoding"
+	"github.com/beatlabs/patron/encoding/json"
+	patronErrors "github.com/beatlabs/patron/errors"
+	"github.com/beatlabs/patron/log"
+	"github.com/opentracing/opentracing-go"
+)
+
```

Comment:
do we really need these constants here ?
The same are in sarama.produce_request.go

Reply:
I see the duplication.    
I tried swapping them out, but patron's API felt cleaner if we can avoid things as `With...(ack sarama.RequiredAcks)` and `sarama.NoResponse`, as the end-user doesn't have to know about Sarama, and I'd like to avoid type casting.

Reply:
cool, makes sense !